tests/test_decode.py

```diff
@@ -135,74 +135,73 @@ def testReadFloat(self):
         
         # 0 length
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'')
+        self.mockStream = BytesIO(b'')
         self.assertEqual(readFloat(self.mockStream, 0), 0.0)
         
         # 4-bit length
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\x00\x00\x00\x00')
+        self.mockStream = BytesIO(b'\x00\x00\x00\x00')
         self.assertEqual(readFloat(self.mockStream, 4), 0.0)
         
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\x3f\x80\x00\x00')
+        self.mockStream = BytesIO(b'\x3f\x80\x00\x00')
         self.assertEqual(readFloat(self.mockStream, 4), 1.0)
         
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\xc0\x00\x00\x00')
+        self.mockStream = BytesIO(b'\xc0\x00\x00\x00')
         self.assertEqual(readFloat(self.mockStream, 4), -2.0)
         
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\x3e\xaa\xaa\xab')
+        self.mockStream = BytesIO(b'\x3e\xaa\xaa\xab')
         self.assertAlmostEqual(readFloat(self.mockStream, 4), 1.0/3.0)
     
         # 8-bit length
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\x00\x00\x00\x00\x00\x00\x00\x00')
+        self.mockStream = BytesIO(b'\x00\x00\x00\x00\x00\x00\x00\x00')
         self.assertEqual(readFloat(self.mockStream, 8), 0.0)
         
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\x3f\xf0\x00\x00\x00\x00\x00\x00')
+        self.mockStream = BytesIO(b'\x3f\xf0\x00\x00\x00\x00\x00\x00')
         self.assertEqual(readFloat(self.mockStream, 8), 1.0)
         
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\xc0\x00\x00\x00\x00\x00\x00\x00')
+        self.mockStream = BytesIO(b'\xc0\x00\x00\x00\x00\x00\x00\x00')
         self.assertEqual(readFloat(self.mockStream, 8), -2.0)
         
         self.mockStream.seek(0)
-        self.mockStream = StringIO(b'\x3f\xd5\x55\x55\x55\x55\x55\x55')
+        self.mockStream = BytesIO(b'\x3f\xd5\x55\x55\x55\x55\x55\x55')
         self.assertEqual(readFloat(self.mockStream, 8), 1.0/3.0)
     
     
     
     def testReadString(self):
         """ Test reading strings. """
         
-        self.mockStream = StringIO(b'')
-        val = readString(self.mockStream, 0)
-        self.assertEqual(val, u'')
+        self.mockStream = BytesIO(b'')
+        self.assertEqual(readString(self.mockStream, 0), b'')
         
-        self.mockStream = StringIO(b'test')
-        val = readString(self.mockStream, 4)
-        self.assertEqual(val, u'test')
+        self.mockStream = BytesIO(b'test')
+        self.assertEqual(readString(self.mockStream, len(self.mockStream.getvalue())),
+                         b'test')
```

Comment:
I think using the temporary variable makes the test a little more readable. The first line is a bit too long.